### PR TITLE
ember flight icon liquid size

### LINKED ISSUE #327: Add "liquid" size to the `ember-flight-icon` addon
In the new Badge component the icon size for the `type=small` variant is `12px`. 

screenshot_297

At the moment our `ember-flight-icon` addon doesn't support this size. We need to introduce a "liquid" size that adapts to the parent container, so we can support also alternative sizes.

_Notice: this is a blocker for https://github.com/hashicorp/design-system-components/pull/1_

/cc @hashicorp/design-systems

### PR DESCRIPTION
**Pull request by didoo**
## :pushpin: Summary

If merged, this PR will fix #327.

## :hammer_and_wrench: Detailed Description

I have:
- added support for the “stretched” prop in the “FlightIcon” Ember addon
- updated the “percy-test” page
- updated the "engineering" documentation
- updated the unit tests

At the beginning I thought to use the `size` parameter, with a `stretch` value, but then I realized we use it to extrapolate the asset name, so it would not work (how do we know in that case which one of the two assets in two different sizes to use?).

/cc @hashicorp/design-systems

### CODE REVIEW

File: ember-flight-icons/addon/components/flight-icon.js
Comment (Dhaulagiri) [inline ember-flight-icons/addon/components/flight-icon.js:51]: The previous logic above defaulted to `true` which meant that by default `flight-icon-display-inline` was added, but with this change that is no longer the case which seems like it would be a breaking change. Is that intentional?

File: ember-flight-icons/tests/dummy/app/templates/percy-test.hbs
Comment (Dhaulagiri) [inline ember-flight-icons/tests/dummy/app/templates/percy-test.hbs:25]:
```suggestion
    <div class="w-8 h-8">
```

File: ember-flight-icons/addon/components/flight-icon.js
Comment (didoo) [reply ember-flight-icons/addon/components/flight-icon.js:51]: yes, nice catch! I have fixed the qunit tests (some were broken, looking for a class that was not there) and added a new one to cover this case. check my new implementation of the `displayClass` method, see if it's OK (but I am open to more elegant solutions, if you have one)

Comment (Dhaulagiri) [reply ember-flight-icons/addon/components/flight-icon.js:51]: This seems ok to me. I'm a little fuzzy on why we don't want to allow inline-block in liquid mode but i'm super rusty here so will defer to your expertise.

**Review by Dhaulagiri — APPROVED** [review]

**Comment by didoo** [comment]
@MelSumner do you mind do a second pass and, if it's OK, approve too? I prefer a double review on this change.

File: ember-flight-icons/addon/components/flight-icon.js
Comment (didoo) [reply ember-flight-icons/addon/components/flight-icon.js:51]: with `display: block`:
screenshot_309

with `display: inline-block`:
screenshot_310

(the pink box is the container)

**Comment by didoo** [comment]
@MelSumner is it ok to merge now?

For future reference, we had a meeting with Melanie and we agreed:
- it's OK to have the classes computed via JS instead of inline in the template, because 
  - this is a more resilient approach for the future
  - when there will be much more complex components, that may need even more (UI) logic to control the class names
  - this approach raises a little bit the "local" complexity, but overall reduces the "systemic" complexity (we have a common pattern adopted for all the components, instead of case by case, so the cognitive load for the maintainers to move from the code of one component to another will be much lower)
- it's ok _for now_ to use "layout containers" (`div.hbs-badge__icon` and `div.hbs-badge__text`) to control the internal layout of the components
  - this "separation of concerns" between _containers_ and _content_ worked extremely well for me in the past
  - we avoid mix&match of class names in the content elements
  - in any case we have set up a meeting in 3/4 weeks in which we will review this decision

/cc @hashicorp/design-systems

**Review by MelSumner — APPROVED** [review]
I don't really understand this, but I don't object so I am not going to block it either. Let's ship it and see how it goes.

**Comment by didoo** [comment]
@MelSumner I am not sure how you can have the Badge working, with an icon size of 12px, without this change (unless you style directly the SVG in the Badge, in which case this for me is an issue, we should not).

**Comment by didoo** [comment]
@MelSumner by design this icon should be 12px in size (as the container), instead is 16px (as per SVG width/height without the `stretched` option we have introduce here). Makes sense?

screenshot_319

**Comment by MelSumner** [comment]
> @MelSumner by design this icon should be 12px in size (as the container), instead is 16px (as per SVG width/height without the `stretched` option we have introduce here). Makes sense?
> 
> screenshot_319

yup makes sense- I figured we'd refactor the badge a little after this is released. 👍

**Pull request merged**
